src/accountlist/accountlistSelectors.js

```diff
@@ -0,0 +1,22 @@
+import { fromJS } from 'immutable';
+
+export const getActiveAccount = (state) =>
+  state.accountlist.get(0);
+
+export const getAuth = (state) => {
+  const account = getActiveAccount(state);
+
+  if (!account) {
+    return fromJS({
```

Comment:
Yeah I think we should avoid deep conversion. For example, that allows us to keep messages in an Immutable array, but the elements themselves can be the raw objects we retrieve from the server (which is more performant)